R/constructNWISURL.R

```diff
@@ -362,27 +346,37 @@ constructWQPURL <- function(siteNumbers,
                             parameterCd,
                             startDate,
                             endDate,
-                            zip = TRUE) {
+                            legacy = FALSE) {
+  
   multiplePcodes <- length(parameterCd) > 1
-  siteNumbers <- paste(siteNumbers, collapse = ";")
-
+  
   if (all(nchar(parameterCd) == 5)) {
     suppressWarnings(pCodeLogic <- all(!is.na(as.numeric(parameterCd))))
   } else {
     pCodeLogic <- FALSE
     parameterCd <- sapply(parameterCd, utils::URLencode, USE.NAMES = FALSE, reserved = TRUE)
   }
-
-  if (multiplePcodes) {
-    parameterCd <- paste(parameterCd, collapse = ";")
+  pcode_name <- ifelse(pCodeLogic, "pCode", "characteristicName")
+  parameterCd <- paste0(pcode_name, "=", parameterCd)
+  
+  if(legacy){
+    if (multiplePcodes) {
+      parameterCd <- paste(parameterCd, collapse = ";")
+    }
+    
+    siteNumbers <- paste(siteNumbers, collapse = ";")
+    baseURL <- drURL("Result", siteid = siteNumbers, Access = pkg.env$access)
+  } else {
+    if (multiplePcodes) {
+      parameterCd <- paste0(parameterCd, collapse = "&")
+    } 
+    
+    siteNumbers <- paste(paste0("siteid=", siteNumbers), collapse = "&")
+    baseURL <- drURL("WQX3", Access = pkg.env$access)
+    baseURL <- paste0(baseURL, siteNumbers)
   }
-
-  baseURL <- drURL("Result", siteid = siteNumbers, Access = pkg.env$access)
-  url <- paste0(
-    baseURL,
-    ifelse(pCodeLogic, "&pCode=", "&characteristicName="),
-    parameterCd
-  )
+  
+  url <- paste0(baseURL, "&", parameterCd)
```

Comment:
at one point, I was trying to keep the "baseURL" to be what we'd use with a POST, and the "url" to be the full url that we'd use with a GET. But that's all gone now, so it can just be 1 variable. Going to WQX3 services though, we do need to use tat paste instead of the drURL because WQP won't take semicolons anymore